bob/src/cluster/quorum.rs

```diff
@@ -221,23 +229,83 @@ impl Quorum {
         debug!("need additional local alien copies: {}", failed_nodes.len());
         let vdisk_id = self.mapper.vdisk_id_from_key(key);
         let operation = Operation::new_alien(vdisk_id);
-        let local_put = put_local_all(
+        let local_put =
+            put_local_all(&self.backend, failed_nodes.clone(), key, data, operation).await;
+        if let Err(e) = local_put {
+            error!(
+                "PUT[{}] local put failed, smth wrong with backend: {:?}",
+                key, e
+            );
+            Err(Error::internal())
+        } else {
+            Ok(())
+        }
+    }
+
+    fn get_secondary_support_nodes(&self, primary_sup_nodes: &Vec<&Node>) -> Vec<&Node> {
+        let secondary_sup_nodes: Vec<_> = self
+            .mapper
+            .nodes()
+            .values()
+            .filter(|n| !primary_sup_nodes.contains(n) && n.name() != self.mapper.local_node_name())
+            .collect();
+        secondary_sup_nodes
+    }
+
+    pub(crate) async fn delete_aliens(
+        &self,
+        mut failed_nodes: Vec<String>,
+        key: BobKey,
+        timestamp: u64,
+    ) -> Result<(), Error> {
+        trace!("selection of free nodes available for data writing");
+        let all_nodes: Vec<_> = self.mapper.nodes().values().collect();
+        let desk: Vec<_> = all_nodes
+            .iter()
+            .flat_map(|n| {
+                all_nodes
+                    .iter()
+                    .filter(|n1| n.name() != n1.name())
+                    .map(move |n1| (n, n1.name().to_string()))
+            })
+            .collect();
+        debug!("DELETE[{}]", key);
+        let queries: Vec<_> = desk
```

Comment:
This gives n^2 queries. This is insane.

Proper logic:
1. Call `mapper.get_support_nodes(..)` to get support nodes for failed ones
2. Send alien delete request to that nodes, failed nodes are passed to `force_alien_nodes` one to each node (logic are the same as in `put_aliens`) 
3. Get list of all nodes from `mapper` and filter out local node and already processed support ones
4. Send alien delete request to that nodes (`force_alien_nodes` is empty)
5. Run delete on local aliens. If there is still some unprocessed failed nodes, then they are put to `force_alien_nodes` list (same idea as in `put_aliens`)